app/main.py

```diff
@@ -46,12 +47,15 @@
 
 
 @app.on_event("startup")
-async def start_prometheus_metrics():
+async def start_service():
     """setup metrics on startup"""
     # condition instrumentation by FACETS_ENABLE_METRICS
     Instrumentator(should_respect_env_var=True, env_var_name="FACETS_ENABLE_METRICS").instrument(
         app
-    ).expose(app)
+    ).expose(app),
+    # Clearing cache and refetching data-quality
+    # Refetching data every hour
+    global_quality_refresh
```

Comment:
this does no work, you are just referencing a function …

Also you can have more than one function that use `@app.on_event("startup")` (that's one of the benefit of events, there are multiple listeners/handlers).

So you should better write it like this:

```suggestion

@app.on_event("startup")
async def start_global_quality_refresh():
    # Clearing cache and refetching data-quality
    # Refetching data every hour
    global_quality_refresh()
```

Reply:
Oh cool , I didn't knew we can use multiple "app.on_event" that's why I use that way :-)

Reply:
But I wonder how its works in local 🤔

Reply:
@alexgarel updated!